connect-inject/container_sidecar.go

```diff
@@ -1,12 +1,42 @@
 package connectinject
 
 import (
+	"bytes"
 	"strings"
+	"text/template"
 
 	corev1 "k8s.io/api/core/v1"
 )
 
-func (h *Handler) containerSidecar(pod *corev1.Pod) corev1.Container {
+func (h *Handler) containerSidecar(pod *corev1.Pod) (corev1.Container, error) {
+	// Create expected volume mounts
+	volMounts := []corev1.VolumeMount{
+		corev1.VolumeMount{
+			Name:      volumeName,
+			MountPath: "/consul/connect-inject",
+		},
+	}
+
+	if h.AuthMethod != "" {
```

Comment:
the only thing that needs the service account mount point is the init container doing `consul login`. I think you can omit this explicit mount here.

Reply:
Removed the sidecar volume mount copying.